src/gui/elevation/qgselevationprofilecanvas.cpp

```diff
@@ -344,6 +344,24 @@ void QgsElevationProfileCanvas::scalePlot( double factor )
   scalePlot( factor, factor );
 }
 
+QgsPointXY QgsElevationProfileCanvas::snapToPlot( QPoint point )
+{
+  if ( !mCurrentJob || !mSnappingEnabled )
+    return QgsPointXY();
+
+  const QgsPointXY plotPoint = canvasPointToPlotPoint( point );
+
+  const double toleranceInPixels = QFontMetrics( font() ).horizontalAdvance( ' ' );
+  const double xToleranceInPlotUnits = 2 * ( mPlotItem->xMaximum() - mPlotItem->xMinimum() ) / ( mPlotItem->plotArea().width() ) * toleranceInPixels;
+  const double yToleranceInPlotUnits = 10 * ( mPlotItem->yMaximum() - mPlotItem->yMinimum() ) / ( mPlotItem->plotArea().height() ) * toleranceInPixels;
```

Comment:
What is the rationale that X and Y have different tolerance in pixels? I would probably expect the same tolerance in both directions...

Reply:
Again, just based on my experimentation with how the snapping "feels" in action. Small horizontal snap distances feels right, yet small vertical snap feels unhelpful. I think partly this may be a result of how the current profile tool works, where vertical snap is always happening, regardless of how far the mouse cursor is from the surface. (So it's potentially a mid-term approach while we ween people off the plugin)